library/huggingface_util.py

```diff
@@ -0,0 +1,71 @@
+from typing import *
+from huggingface_hub import HfApi
+from pathlib import Path
+import argparse
+import os
+
+from library.utils import fire_in_thread
+
+
+def exists_repo(
+    repo_id: str, repo_type: str, revision: str = "main", token: str = None
+):
+    api = HfApi(
+        token=token,
+    )
+    try:
+        api.repo_info(repo_id=repo_id, revision=revision, repo_type=repo_type)
+        return True
+    except:
+        return False
+
+
+@fire_in_thread
```

Comment:
なるほど、理解しました。
しかし大規模な学習の際にはstateをアップロードするのを待つとかなり時間のロスになるので、「デフォルトで同期、オプションで非同期にする」といった仕様で大丈夫でしょうか？
最後の保存の際は同期になるように修正します！

---

Okay, I understand.
However, in the case of large-scale learning, waiting for the state to be uploaded will waste a lot of time, so is it okay to use a specification such as "sync by default, async as an option"?
I will fix it so that the last save will be in sync!